runhouse/resources/module.py

```diff
@@ -19,71 +19,32 @@
 logger = logging.getLogger(__name__)
 
 # These are methods that the Module's __getattribute__ logic should not intercept to run remotely
-LOCAL_METHODS = [
-    "RESOURCE_TYPE",
-    "__class__",
-    "__delattr__",
-    "__dict__",
-    "__dir__",
-    "__doc__",
-    "__eq__",
-    "__format__",
-    "__ge__",
-    "__gt__",
-    "__hash__",
-    "__init__",
-    "__init_subclass__",
-    "__le__",
-    "__lt__",
-    "__module__",
-    "__ne__",
-    "__new__",
-    "__reduce__",
-    "__reduce_ex__",
-    "__repr__",
-    "__setattr__",
-    "set_async",
-    "__getattribute__",
-    "__sizeof__",
-    "__str__",
-    "__subclasshook__",
-    "__weakref__",
-    "_check_for_child_configs",
-    "_cls_pointers",
+# Should we just be using inspect.getclasstree to get signatures without the Module methods instead?
+LOCAL_METHODS = dir(Resource) + [
```

Comment:
what happens if a user creates a class where they inherit `rh.Module` and have a method `env` on their class?

Reply:
It overrides the built-in property (which __getattribute__ will handle properly), but is still run locally rather than as an RPC if the system is a remote cluster. You can see this behavior in function.py, which overrides a bunch of things, most notably `.to`.